### PR TITLE
feat(terraform): Check added for AWS Database instance deletion protection

### LINKED ISSUE #1083: CKV_AWS_139 - RDS cluster deletion protection
**Is your feature request related to a problem? Please describe.**
I think it is great to check, if deletion protection is enabled or not, but is there a reason, why it is only checking for `aws_rds_cluster` and not includes `aws_db_instance`? They are quite similar.

I'm happy to either extend this check to include DB instances or add a new one to check separately for them. 

**Additional context**
https://github.com/bridgecrewio/checkov/blob/caafcbb83d6901b57c9a39464673670c150f0a33/checkov/terraform/checks/resource/aws/RDSDeletionProtection.py#L7-L15

### PR DESCRIPTION
**Pull request by Jhanvi12**
* Added Rule ID is CKV_AWS_293
* added test case for same.

**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:       to indicate new features or checks
    - fix:         to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description
This policy CKV_AWS_293 to ensure deletion protection for AWS Database instance . For that  test case has also been pushed .

Fixes #1083 

Deletion protection should be set to true to fix this .
**/**Sample Terraform**/**
```
resource "aws_database_instance" "default-tf" {
  + deletion_protection = true
}
```

### Description
deletion_protection is not set to `true` for aws_database_instance

### Fix
set deletion_protection to `true` as shown in the sample terraform

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by JamesWoolfenden — APPROVED** [review]
LGTM

File: checkov/terraform/checks/resource/aws/RDSInstanceDeletionProtection.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from checkov.terraform.checks.resource.base_resource_value_check import BaseResourceValueCheck
+from checkov.common.models.enums import CheckCategories
+
+
+class RDSInstanceDeletionProtection(BaseResourceValueCheck):
+
+    def __init__(self):
+        name = "Ensure that AWS database instance have deletion protection enabled"
```
Comment (gruebel) [inline checkov/terraform/checks/resource/aws/RDSInstanceDeletionProtection.py:8]:
```suggestion
        name = "Ensure that AWS database instances have deletion protection enabled"
```
or
```suggestion
        name = "Ensure that AWS database instance has deletion protection enabled"
```

**Review by gruebel — COMMENTED** [review]
nice work, just a small change and then it is good to merge

File: checkov/terraform/checks/resource/aws/RDSInstanceDeletionProtection.py (hunk 1 above)
Comment (Jhanvi12) [reply checkov/terraform/checks/resource/aws/RDSInstanceDeletionProtection.py]: I have pushed the said changes as  name = "Ensure that AWS database instances have deletion protection enabled" . Let me know if more information required ! :)

**Review by gruebel — APPROVED** [review]
thanks, looks good 💪

**Pull request merged**